### DISCUSSION TITLE
Links appear to be clickable but do not open browser

### DISCUSSION THREAD

**Discussion by jasonlong** [wezterm/wezterm#3672] [General]
I'm not sure if there's a config option I'm missing, but when I hover over links in wezterm they underline and my cursor changes to the correct hand cursor. However, when I click any link nothing at all happens.

Is there something I need to configure in wezterm or in macOS?

**Comment by wez** [comment]
wezterm logs a line to the log (which you can see in the Debug Overlay), and then runs `/usr/bin/open -a URL` when you click on a link on macOS.

**Comment by jasonlong** [comment]
Oh maybe this is the problem then – it looks like `open` should be using the `-u` option.

On my system (macOS 13.1.1):

`-a: Opens with the specified application.`
`-u: Open this URL, even if it matches exactly a filepath`

**Reply by wez** [reply]
Sorry, I quoted the wrong bit of code:

https://github.com/wez/wezterm/blob/b44dce00f229bf5367d450d3ac4881f959bb5d88/wezterm-open-url/src/lib.rs#L10

clicking links works for me on macos.

**Reply by jasonlong** [reply]
Ah ok. So it must be something on my end. I'll keep investigating - thank you for your help. 🙇🏼

**Reply by bobsongplus** [reply]
@jasonlong  Hi, have you solved your problem? I have encountered the same problem,  but I can use the command `/usr/bin/open` URL.

**Reply by jasonlong** [reply]
@bobsongplus Same for me. `open` works on the command line, but clicking links in wezterm doesn't do anything.

**Reply by PtrZjc** [reply]
Googling of the same problem brought me here. Please share a solution if you find it.

**Comment by bzvestey** [comment]
Just dropping this here since I was having the same issue. For me the issue wasn't with wezterm, but the fact that I was actually using zellij inside of it. In zellij I needed to hold shift while clicking the link to open it. There might be something else you are running that could be changing this interaction, and would probably be worth a check.

**Reply by anthony-halim** [reply]
May not be related since I'm not running MacOS, but upvoting above since this also works for Ubuntu 22.04 OS.

Running Zellij inside Wezterm requires shift + click to open a link. Running not inside Zellij (but still within Wezterm) requires double click to open the link.

**Reply by W3ND31** [reply]
For me I was having this problem with Lazygit inside nvim and holding shift worked! Thx @bzvestey

**Reply by pahMelnik** [reply]
Thx it's work for me, i am use tmux

**Reply by Subterrane** [reply]
I had the same problem, found this comment, recognized the user icon before the name and realized I used to work with @bzvestey. I tried ctrl+click & cmd+click but was too dumb to try shift+click. Thanks, Bryan!

**Reply by Bhupesh-V** [reply]
Same goes for Neovim.

**Comment by jasonlong** [comment]
The comment from @anthony-halim above about Zellij made me realize this could be a tmux issue.

I tested in a non-tmux Wezterm window and I can indeed click on links to open just fine. Inside tmux, <kbd>Shift</kbd>+click does the trick. I know I tried other modifiers in the past, but apparently never tried <kbd>Shift</kbd>.

As the author of this thread, I'm going to close this. Someone else can reopen if they like.

**Reply by duarteocarmo** [reply]
Upvote this for the love of god!

**Reply by mvillafuertem** [reply]
@wez @jasonlong @duarteocarmo  why this keybinding doesn't work for me inside tmux...inside tmux i have to use SHIFT+Click

```lua
  config.mouse_bindings = {
    -- CMD-click will open the link under the mouse cursor
    {
      event = { Up = { streak = 1, button = 'Left' } },
      mods = 'SHIFT',
      action = wezterm.action.DisableDefaultAssignment,
    },
    {
      event = { Up = { streak = 1, button = 'Left' } },
      mods = 'CMD',
      action = wezterm.action.OpenLinkAtMouseCursor,
    },
  }
```

**Reply by deanc** [reply]
Same issue with tmux here. Hopefully we can find a solution in the future but shift+click muscle memory will do the trick!

**Reply by joshuataylor** [reply]
try
```lua
        action = wezterm.action.Multiple {
            wezterm.action.OpenLinkAtMouseCursor,
            wezterm.action.Nop
        }
```

example mine for command click
```lua
    {
        event = { Up = { streak = 1, button = "Left" } },
        mods = "NONE",
        action = wezterm.action.DisableDefaultAssignment,
    },

    {
        event = { Up = { streak = 1, button = "Left" } },
        mods = "CMD",
        action = wezterm.action.Multiple {
            wezterm.action.OpenLinkAtMouseCursor,
            wezterm.action.Nop
        }
    },
```

**Comment by bruno-notifi** [comment]
Same issue

**Comment by ManuGraiph** [comment]
I'm experiencing the exact same behaviour but it doesn't seem to be related to either `tmux` or `wezterm`. I have issues with `khal` on the interactive mode, i'm testing two terminals (kitty and wezterm), with bash, fish and zsh and in all the possible combinations i have the exact same issue, can't click links but i can shift-click them (thx for the answer pointing that, it saved me after hours of google). I'm not using tmux or screen or any other multiplexer.

Any idea why this happens? It's possible to change this behaviour for the links or it's app-related?

**Comment by db757** [comment]
Same issue on windows, works when I double click though.

**Comment by vermdeep** [comment]
It was not working for me on MacOS 14.5. It worked after I added following mouse-bindings:
```lua
config.mouse_bindings = {
	-- CMD-click will open the link under the mouse cursor
	{
		event = { Up = { streak = 1, button = "Left" } },
		mods = "SUPER",
		action = wezterm.action.OpenLinkAtMouseCursor,
	},
}
```

* It only worked if I Launched wezterm from any other window.
* However, After adding dummy `wezterm.on` handler for `open-uri` event, it worked for directly launched wezterm process as well.
```
wezterm.on("open-uri", function(window, pane, uri)
	wezterm.log_info(window)
	wezterm.log_info(pane)
	wezterm.log_info(uri)
end)
```

**Comment by sankarraj-ps** [comment]
I have the same issue, even when not using `tmux`

**Reply by p53** [reply]
I am on MacOS i was also scratching my head with this, trying different keybindings, until i realized that i have to click on underline in link, otherwise it won't open, so you have to be pretty precise in clicking directly on underline...quite confusing

**Comment by ravihlb** [comment]
Using `tmux` on MacOS 15, opened with double click.

**Reply by zzeroo** [reply]
On linux, within tmux it is SHIFT+double click
